File: salesforge/frontend/src/types/deal.ts
export type DealStage = 'prospect' | 'proposal' | 'negotiation' | 'won' | 'lost'

export const DEAL_STAGES: DealStage[] = ['prospect', 'proposal', 'negotiation', 'won', 'lost']

export const STAGE_DEFAULT_PROBABILITY: Record<DealStage, number> = {
  prospect: 10,
  proposal: 30,
  negotiation: 60,
  won: 100,
  lost: 0,
}

export interface Deal {
  id: string
  title: string
  value: number | null
  currency: string
  stage: DealStage
  probability: number
  expected_close_date: string | null
  contact_id: string | null
  company_id: string | null
  owner_id: string
  created_at: string
  updated_at: string
}

export interface DealListResponse {
  items: Deal[]
  total: number
  page: number
  page_size: number
}

export interface PipelineStageSummary {
  stage: DealStage
  count: number
  total_value: number
}

export interface DealCreate {
  title: string
  value?: number
  currency?: string
  stage?: DealStage
  probability?: number
  expected_close_date?: string
  contact_id?: string
  company_id?: string
}

export type DealUpdate = Partial<DealCreate>

File: salesforge/backend/app/schemas/deal.py
from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class DealCreate(BaseModel):
    title: str
    value: float | None = None
    currency: str = "USD"
    stage: str = "prospect"
    probability: int | None = None
    expected_close_date: date | None = None
    contact_id: str | None = None
    company_id: str | None = None


class DealUpdate(BaseModel):
    title: str | None = None
    value: float | None = None
    currency: str | None = None
    stage: str | None = None
    probability: int | None = None
    expected_close_date: date | None = None
    contact_id: str | None = None
    company_id: str | None = None


class DealResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    title: str
    value: float | None
    currency: str
    stage: str
    probability: int
    expected_close_date: date | None
    contact_id: str | None
    company_id: str | None
    owner_id: str
    created_at: datetime
    updated_at: datetime


class DealListResponse(BaseModel):
    items: list[DealResponse]
    total: int
    page: int
    page_size: int


class PipelineStageSummary(BaseModel):
    stage: str
    count: int
    total_value: float
